### PR TITLE
Add support for user-events monitoring

### LINKED ISSUE #519: Add support for user-events monitoring
Add the [user-events](https://github.com/adobe-apiplatform/openwhisk-user-events) service as part the deployment. This will provide monitoring capabilities for all invocations passing through the cluster. More on the [supported events](https://github.com/apache/openwhisk/blob/master/docs/metrics.md#supported-events).
Additionally deploy `Prometheus` and `Grafana` for storing and visualizing the metrics using predefined dashboards.
These services should only be installed if the `userMetricsEnabled` is set to true: 
```
metrics:
  userMetricsEnabled: true
```
Having a `Prometheus` deployed in the cluster allows also the scraping of `controller` and `invoker` metrics. 
Enable metrics scraping on Prometheus by setting `prometheusEnabled` to `true`: 
```
metrics:
  prometheusEnabled: true
```

### PR DESCRIPTION
**Pull request by selfxp**
* Provides monitoring capabilities for all user specific metrics by deploying `user-events`, `prometheus` and `grafana` on the cluster in case `userMetricsEnabled` is set to `true`. 

* It exposes a `/monitoring` endpoint for accessing the Grafana dashboards. 

* All system metrics will also be stored inside prometheus in case `prometheusEnabled=true`. 

* More on the `user-events` service that reads the events from Kafka on this [OW PR](https://github.com/apache/openwhisk/pull/4584).

Fixes #519

### CODE REVIEW

**Comment by dgrove-oss** [comment]
This looks great!   Took a quick scan; will go through more carefully next week.

TravisCI is failing because the prometheus-server pod is going into a crash loop backoff.  Unfortunately the upload of logs from TravisCI to box hasn't been working for a while, so if you can't recreate locally you may need to debug by hacking up the script that runs the test to do some additional `kubectl` operations to figure out why that pod is sad.

**Comment by selfxp** [comment]
Thx for the tip @dgrove-oss, I was able to reproduce the problem locally. Prometheus deploy should be working fine now with or without persistence.

File: helm/openwhisk/templates/nginx-cm.yaml [hunk 1]
```diff
@@ -71,6 +71,7 @@ data:
         set $controllers {{ include "openwhisk.controller_host" . }};
 {{- if or (eq .Values.whisk.ingress.type "NodePort") (eq .Values.whisk.ingress.type "LoadBalancer") }}
         set $apigw {{ include "openwhisk.apigw_host" . }};
+        set $grafana {{ include "openwhisk.grafana_host" . }};
```
Comment (dgrove-oss) [inline helm/openwhisk/templates/nginx-cm.yaml]: Is it ok if `openwhisk.grafana_host` doesn't resolve to a host that actually exists (user metrics not enabled)?  Or should this and/or the stanza below adding `/monitoring` both be inside a 
`{{ if or .Values.metrics.prometheusEnabled .Values.metrics.userMetricsEnabled }}`
check?

File: helm/openwhisk/values.yaml [hunk 2]
```diff
@@ -307,6 +307,42 @@ redis:
...
+  imageTag: "6.3.0"
+  imagePullPolicy: "IfNotPresent"
+  replicaCount: 1
+  restartPolicy: "Always"
+  port: 3000
+  adminPassword: "admin"
+  dashboards:
+  - https://raw.githubusercontent.com/adobe-apiplatform/openwhisk-user-events/master/compose/grafana/dashboards/openwhisk_events.json
+  - https://raw.githubusercontent.com/adobe-apiplatform/openwhisk-user-events/master/compose/grafana/dashboards/global-metrics.json
+  - https://raw.githubusercontent.com/adobe-apiplatform/openwhisk-user-events/master/compose/grafana/dashboards/top-namespaces.json
```
Comment (dgrove-oss) [inline helm/openwhisk/values.yaml]: is usage of adobe-apiplatform organization for the git content temporary staging or permanent?

File: helm/openwhisk/templates/nginx-cm.yaml (hunk 1 above)
Comment (selfxp) [reply helm/openwhisk/templates/nginx-cm.yaml]: Yes, good point, a miss on my part, will check if `prometheus` or `user-metrics` is enabled.

File: helm/openwhisk/values.yaml (hunk 2 above)
Comment (selfxp) [reply helm/openwhisk/values.yaml]: Temporary until [#4584](https://github.com/apache/openwhisk/pull/4584) is merged

**Comment by dgrove-oss** [comment]
@selfxp -- should we go ahead and merge this now?

**Comment by selfxp** [comment]
@dgrove-oss yes please, it has all the discussed changes in place

**Pull request merged**